_includes/releases/v21.2/v21.2.7.md

```diff
@@ -0,0 +1,86 @@
+- Fixed a bug where a [`RESTORE`](../v21.2/restore.html) job could hang if it encountered an error when ingesting restored data. [#76509][#76509]
+- Fixed a race condition that in rare circumstances could cause a node to panic with `unexpected Stopped processor` during shutdown. [#76827][#76827]
+- There is now a 1 hour timeout when sending [Raft](../v21.2/architecture/replication-layer.html#raft) snapshots, to avoid stalled snapshot transfers preventing Raft log truncation and growing the Raft log very large. This is configurable via the `COCKROACH_RAFT_SEND_SNAPSHOT_TIMEOUT` environment variable. [#76829][#76829]
+- Fixed an error that could sometimes occur when sorting the output of the [`SHOW CREATE ALL TABLES`](../v21.2/show-create.html) statement. [#76698][#76698]
+- Fixed a bug where [`CASE` expressions](../v21.2/scalar-expressions.html#conditional-expressions) with branches that result in types that cannot be cast to a common type caused internal errors. They now result in a user-facing error. [#76616][#76616]
+- Error messages produced during import are now truncated. Previously, [`IMPORT`](../v21.2/import.html) could potentially generate large error messages that could not be persisted to the jobs table, resulting in a failed import never entering the failed state and instead retrying repeatedly. [#76980][#76980]
+- Fixed a bug that could corrupt indexes containing [virtual columns](../v21.2/computed-columns.html) or [expressions](../v21.2/expression-indexes.html). The bug only occurred when the index's table had a foreign key reference to another table with an `ON DELETE CASCADE` action, and a row was deleted in the referenced table. This bug was present since virtual columns were added in version v21.1.0. [#77053][#77053]
+- [Changefeeds](../v21.2/changefeed-sinks.html) retry instead of fail on RPC send failure. [#77069][#77069]
+- Fixed a bug that caused the **Open Transactions** chart on the [**Metrics**](../v21.2/ui-overview-dashboard.html) page to constantly increase for empty transactions. [#77236][#77236]
+- Fixed a bug that could interfere with a system table migration. [#77309][#77309]
+- The content type header for the HTTP log sink is now set to `application/json` if the format of the log output is `JSON`. [#77341][#77341]
+- Fixed a bug where draining nodes in a cluster without shutting them down could stall foreground traffic in the cluster. [#77490][#77490]
+
+<h3 id="v21-2-7-performance-improvements">Performance improvements</h3>
+
+- Fixed a bug in the histogram estimation code that could cause the optimizer to think a scan of a multi-column index would produce 0 rows, when in fact it would produce many rows. This could cause the optimizer to choose a suboptimal plan. It is now less likely for the optimizer to choose a suboptimal plan when multiple multi-column indexes are available. [#76555][#76555]
```

Comment:
```suggestion
- Fixed a bug in the histogram estimation code that could cause the [optimizer](../v21.2/cost-based-optimizer.html) to think a scan of a multi-column index would produce 0 rows, when in fact it would produce many rows. This could cause the optimizer to choose a suboptimal plan. It is now less likely for the optimizer to choose a suboptimal plan when multiple multi-column indexes are available. [#76555][#76555]
```